p2p/libp2p/p2pSigner.go

```diff
@@ -0,0 +1,42 @@
+package libp2p
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	libp2pCrypto "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type p2pSigner struct {
+	privateKey *libp2pCrypto.Secp256k1PrivateKey
+}
+
+// Sign will sign a payload with the internal private key
+func (signer *p2pSigner) Sign(payload []byte) ([]byte, error) {
+	return signer.privateKey.Sign(payload)
```

Comment:
perhaps a check for privateKey not nil should be added here

Reply:
already mentioned by Bogdan

Reply:
no, we generally do not check for the internal components over and over again for nil checks as this will degrade the performance.